include/boost/geometry/algorithms/detail/counting.hpp

```diff
@@ -84,15 +84,12 @@ template <typename SingleCount>
 struct multi_count
 {
     template <typename MultiGeometry>
-    static inline std::size_t apply(MultiGeometry const& geometry)
+    static inline std::size_t apply(MultiGeometry const& multi)
     {
         std::size_t n = 0;
-        for (typename boost::range_iterator<MultiGeometry const>::type
-                 it = boost::begin(geometry);
-             it != boost::end(geometry);
-             ++it)
+        for (auto const& geometry : multi)
```

Comment:
Range-based for loop is probably incorrect here.

Reply:
:heavy_check_mark: right, reverted